sirepo/raydata/scan_monitor.py

```diff
@@ -254,8 +255,15 @@ async def _incoming(self, body):
         return getattr(self, "_request_" + body.method)(body.data.get("args"))
 
     async def post(self):
+        self._rs_authenticate()
```

Comment:
I wonder if AuthHeaderRequestHandler.post shouldn't always authenticate, but maybe that breaks other uses.

Reply:
I think that could work. Every use eventually calls _rs_authenticate but then do their own handling to validate that the token is what they expect it to be. 

Can you describe / give pseudocode for what you have in mind? Should we use the pattern we have for server api's with deny by default, enforce decoration of all routes, and use an init_api like structure to check?

Reply:
I think it should simply be tornado.AuthRequestHandler and it should have two methods: get and post. get might always raise Forbidden by default. post would look like:

```py
def post(self):
    self.__authenticate()
    self._rs_post()
```

Basically using a dispatch style so the default is always to authenticate. The subclass would implement the _rs methods. __authenticate would call _rs_authenticate and pass the header (token) to _rs_authenticate. If the subclass didn't implement either _rs* method, this class would raise a server error.

Reply:
322a67240